MadDonkeySoftware/corp-hq
# CHQ-99 Validating auth token on applicable requests

### Story
https://maddonkeysoftware.myjetbrains.com/youtrack/issue/CHQ-99

### Summary
* Added middleware to API that validates the `auth-token` header on requests
  * This is enabled by default and must be opted out. See lines 20 through 30-ish for details
* Fixed a bug in `behave-ci-cd.sh` where the test container was removed before the exit code could be captured
* Removed some dead code that was added by a copy / paste mistake
* Fixed a typo of "endpoint"
* Added a explicit data volume for the test mongo instance as that is supposed to improve performance.

### Testing
* Good ol' `behave-ci-cd.sh`
* Start UI, API and Runner projects
* Verify the following:
  * Log in
    * NOTE: you may want to open your browsers developer tools and note the auth token returned after a successful login. You'll need this token to test kicking a job off below.
  * Log out
  * Register
* Verify jobs can still be started
  * curl -X POST http://localhost:5000/api/v1/job -H 'Content-Type: application/json' -H 'auth-token: xxxxxxxxxx' -d '{ "jobType": "ImportMapData" }'
